packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -73,13 +88,24 @@ class RenderParagraph extends RenderBox {
          locale: locale,
          strutStyle: strutStyle,
          textWidthBasis: textWidthBasis,
-       );
+       ) {
+    addAll(children);
+    _extractPlaceholderSpans(text);
+  }
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! TextParentData)
+      child.parentData = TextParentData();
+  }
 
   final TextPainter _textPainter;
+  bool _needsLayout = true;
+
 
   /// The text to display
-  TextSpan get text => _textPainter.text;
-  set text(TextSpan value) {
+  InlineSpan get text => _textPainter.text;
+  set text(InlineSpan value) {
```

Comment:
Do we need to re-extract the placeholder spans if somebody sets a new InlineSpan as text?

Reply:
Yes, good catch!